src/main/java/org/matsim/run/RunKelheimScenario.java

```diff
@@ -2,129 +2,172 @@
+                    .setOpeningTime(8. * 3600.).setClosingTime(20. * 3600.));
+        }
+
+        config.planCalcScore().addActivityParams(new PlanCalcScoreConfigGroup.ActivityParams("car interaction").setTypicalDuration(60));
+        config.planCalcScore().addActivityParams(new PlanCalcScoreConfigGroup.ActivityParams("other").setTypicalDuration(600 * 3));
+
+        config.planCalcScore().addActivityParams(new PlanCalcScoreConfigGroup.ActivityParams("freight_start").setTypicalDuration(60 * 15));
+        config.planCalcScore().addActivityParams(new PlanCalcScoreConfigGroup.ActivityParams("freight_end").setTypicalDuration(60 * 15));
+
+        config.controler().setOutputDirectory(sample.adjustName(config.controler().getOutputDirectory()));
+        config.plans().setInputFile(sample.adjustName(config.plans().getInputFile()));
+
+        config.qsim().setFlowCapFactor(sample.getSize() / 100.0);
+        config.qsim().setStorageCapFactor(sample.getSize() / 100.0);
+
+        config.vspExperimental().setVspDefaultsCheckingLevel(VspExperimentalConfigGroup.VspDefaultsCheckingLevel.info);
+        config.plansCalcRoute().setAccessEgressType(PlansCalcRouteConfigGroup.AccessEgressType.accessEgressModeToLink);
+
+        if (drt) {
+            MultiModeDrtConfigGroup multiModeDrtConfig = ConfigUtils.addOrGetModule(config, MultiModeDrtConfigGroup.class);
+            ConfigUtils.addOrGetModule(config, DvrpConfigGroup.class);
+            DrtConfigs.adjustMultiModeDrtConfig(multiModeDrtConfig, config.planCalcScore(), config.plansCalcRoute());
+        }
+
+        return config;
+    }
+
+    @Override
+    protected void prepareScenario(Scenario scenario) {
+
+        for (Link link : scenario.getNetwork().getLinks().values()) {
+            Set<String> modes = link.getAllowedModes();
+
+            // allow freight traffic together with cars
+            if (modes.contains("car")) {
+                HashSet<String> newModes = Sets.newHashSet(modes);
+                newModes.add("freight");
+
+                link.setAllowedModes(newModes);
+            }
+        }
+
+        if (drt) {
+            scenario.getPopulation()
+                    .getFactory()
+                    .getRouteFactories()
+                    .setRouteFactory(DrtRoute.class, new DrtRouteFactory());
+        }
+
+    }
+
+    @Override
+    protected void prepareControler(Controler controler) {
+        controler.addOverridingModule(new AbstractModule() {
+            @Override
+            public void install() {
+                install(new SwissRailRaptorModule());
+                bind(AnalysisMainModeIdentifier.class).to(KelheimMainModeIdentifier.class);
+                addControlerListenerBinding().to(ModeChoiceCoverageControlerListener.class);
+            }
+        });
+
+        if (drt) {
+            Config config = controler.getConfig();
+            Scenario scenario = ScenarioUtils.loadScenario(config);
```

Comment:
Shouldn't you have access to the scenario via the controler without loading it again?

Reply:
Yeah, that'a good point. Now I am use the controler.getScenario() to get the scenario.